mcdc/loop.py

```diff
@@ -24,13 +27,16 @@
 @njit
 def loop_main(mcdc):
     simulation_end = False
+
+    loop_index = numba.uint64(0)
     while not simulation_end:
+        cycle_seed = kernel.int_hash_combo(loop_index, mcdc["rng_seed"])
         # Loop over source particles
-        loop_source(mcdc)
+        loop_source(cycle_seed, mcdc)
 
         # Loop over source precursors
         if mcdc["bank_precursor"]["size"] > 0:
-            loop_source_precursor(mcdc)
+            loop_source_precursor(cycle_seed, mcdc)
```

Comment:
We are going to need to carefully treat this one.

Reply:
Each argument now gets its own, unique hash. Convert the hex to ASCII for a fun surprise.